docs/command_line_reference.rst

```diff
@@ -682,6 +688,63 @@ Client certificates can be presented regardless of the ``verify_certs`` setting,
 * Enable SSL, verify server certificates using private CA: ``--client-options="use_ssl:true,verify_certs:true,ca_certs:'/path/to/cacert.pem'"``
 * Enable SSL, verify server certificates using private CA, present client certificates: ``--client-options="use_ssl:true,verify_certs:true,ca_certs:'/path/to/cacert.pem',client_cert:'/path/to/client_cert.pem',client_key:'/path/to/client_key.pem'"``
 
+**Static Responses**
+
+Define a JSON file containing a list of objects with the following properties:
+
+* ``path``: A path or path pattern that should be matched. Only leading and trailing wildcards (``*``) are supported. A path containing only a wildcard acts matches any path.
+* ``body``: The respective response body.
+* ``body-encoding``: Either ``raw`` or ``json``. Use ``json`` by default and ``raw`` for the operation-type ``bulk`` and ``search``.
+
+Here we define the necessary responses for a track that bulk-indexes data::
+
+    [
+      {
+        "path": "*/_bulk",
+        "body": {
+          "errors": false,
+          "took": 1
+        },
+        "body-encoding": "raw"
+      },
+      {
+        "path": "/_cluster/health*",
+        "body": {
+          "status": "green",
+          "relocating_shards": 0
+        },
+        "body-encoding": "json"
+      },
+      {
+        "path": "/_all/_stats/_all",
+        "body": {
+          "_all": {
+            "total": {
+              "merges": {
+                "current": 0
+              }
+            }
+          }
+        },
+        "body-encoding": "json"
+      },
+      {
+        "path": "*",
+        "body": {},
+        "body-encoding": "json"
+      }
+    ]
+
+.. note::
+   Paths are evaluated from top to bottom. Therefore, place more restrictive paths at the top of the file.
+
+Save the above responses as ``responses.json`` and execute a benchmark as follows::
+
+    esrally race --track=geonames --challenge=append-no-conflicts-index-only --pipeline=benchmark-only --skip-rest-api-check --distribution-version=8.0.0 --client-options="static_responses:'responses.json'"
+
+.. note::
+   Use ``--skip-rest-api-check`` to ensure that Rally skips any probing requests that are not directly related to the benchmark (e.g. checks whether the cluster is up).
```

Comment:
Could we automatically skip rest api checks when `static_responses` is defined?

Reply:
Thanks for the idea! I've pushed a change in https://github.com/elastic/rally/pull/1234/commits/af77d1bd3d670bdee1ae37c07dd0b3af77a4a275. In that change I've also:

* Turned `--skip-rest-api-check` into an undocumented flag again. With the changes in this PR it's also likely that we won't need that flag at all in the future but I'd like to keep it for now.
* Improved logging
* Avoided issuing *any* additional requests (probing for cluster version, telemetry devices) when static responses are enabled.